bindings/rust/src/bindings/mod.rs

```diff
@@ -184,10 +198,17 @@ impl KZGProof {
         Ok(Self { bytes: proof_bytes })
     }
 
-    pub fn to_bytes(&self) -> Bytes48 {
-        let mut bytes = [0; 48];
-        bytes.copy_from_slice(&self.bytes);
-        Bytes48 { bytes }
+    pub fn to_binding_bytes(&self) -> Bytes48 {
+        Bytes48 { bytes: self.bytes }
+    }
+
+    // NOTE: this function is not public to avoid easy conversions between KzgCommitment and
+    //       KzgProof that go via the [`Bytes48`] type. For the same reason, there is no `From`
+    //       implementation between the types.
```

Comment:
Nice doc.

Can you also please document what these two functions do? I'm comfortable with `to_bytes()` functions having no docs, but `to_binding_bytes()` is adding additional context that can be confusing.

Reply:
yeah this is a weird state. In principle I would like to not expose the `Bytes48` type but the facade API already does. So I changed this function's name and made it tests only but then realized it would be needed to satisfy the public API. Happy to revert this change and discuss the possibility of removing `Bytes48` from the public API in a different PR

Reply:
I'm also strongly in favour of using specific types for commitments and proofs(`KzgProof` and `KzgCommitment` introduced in this PR) instead of using the common `Bytes48` for the public interface to the bindings. 
This would give us higher type safety where we don't accidentally replace a proof with a commitment when passing values to the public functions. 
@asn-d6  I remember a conversation sometime back about using `Bytes48` to have it consistent with the other bindings. Not sure if that is still relevant somehow.

Reply:
`Bytes48` signifies an untrusted bytes object that might happen to be a valid KZG proof. Whereas `KZGProof` is a valid KZG proof returned by the KZG library (or used internally in the KZG library).

The distinction between the two types is important to preserve.

I understand that `Bytes48` is not strongly typed and maybe you would have liked to be `Bytes48Proof` and `Bytes48Commitment`.  If this would be useful to lighthouse let's discuss it. But preferably in another PR (and preferably soon since we are getting ready to freeze for the audit).